File: WePeiYang/lostfound/src/main/java/com/twtstudio/tjwhm/lostfound/release/ReleaseContract.java
package com.twtstudio.tjwhm.lostfound.release;

import com.twtstudio.tjwhm.lostfound.base.BaseBean;
import com.twtstudio.tjwhm.lostfound.base.BaseContract;
import com.twtstudio.tjwhm.lostfound.base.CallbackBean;
import com.twtstudio.tjwhm.lostfound.detail.DetailBean;

import java.io.File;
import java.util.Map;

/**
 * Created by tjwhm on 2017/7/6.
 **/

public interface ReleaseContract {
    public interface ReleaseView extends BaseContract.BaseView {
        void successCallBack(BaseBean baseBean);

        void setEditData(DetailBean detailBean);

        void deleteSuccessCallBack();

        void drawRecyclerView(int position);

        void onTypeItemSelected(int position);

    }

    public interface ReleasePresenter extends BaseContract.BasePresenter {


        void uploadReleaseData(Map<String, Object> map, String lostOrFound);

        void uploadReleaseDataWithPic
                (Map<String, Object> map, String lostOrFound, File file);

        void successCallBack(BaseBean baseBean);


        void successEditCallback(BaseBean baseBean);

        void delete(int id);

        void deleteSuccessCallBack(BaseBean baseBean);

        void uploadEditDataWithPic(Map<String, Object> map, String lostOrFound, File file, int id);
    }

}

File: WePeiYang/lostfound2/src/main/java/com/yookiely/lostfond2/release/ReleaseContract.kt
package com.yookiely.lostfond2.release

import com.yookiely.lostfond2.service.DetailData
import com.yookiely.lostfond2.service.MyListDataOrSearchBean
import java.io.File

interface ReleaseContract {

    interface ReleaseView {

        fun successCallBack(beanList: List<MyListDataOrSearchBean>)

        fun failCallBack(message: String)

        fun setEditData(detailData: DetailData)

        fun deleteSuccessCallBack()

        fun drawRecyclerView(position: Int)

        fun onTypeItemSelected(position: Int)
    }

    interface ReleasePresenter {

        fun uploadReleaseData(map: Map<String, Any>, lostOrFound: String)

        fun uploadReleaseDataWithPic(map: Map<String, Any>, lostOrFound: String, listOfFile: MutableList<File?>)

        fun successCallBack(beanList: List<MyListDataOrSearchBean>)

        fun successEditCallBack(beanList: List<MyListDataOrSearchBean>)

        fun failCallback(message: String)

        fun delete(id: Int)

        fun deleteSuccessCallBack()

        fun uploadEditDataWithPic(map: Map<String, Any>, lostOrFound: String, listOfFile: MutableList<File?>, listOfString: MutableList<String?>, id: Int)

        fun loadDetailDataForEdit(id: Int, releaseView: ReleaseView)
    }
}